src/common/error/request-error-factory.ts

```diff
@@ -27,15 +27,40 @@ export default class RequestErrorFactory {
             return 'timeout';
         }
 
-        if (response.body && typeof response.body.type === 'string') {
+        if (this._isStorefrontErrorResponseBody(response.body)) {
+            return 'default';
+        }
+
+        if (this._isInternalErrorResponseBody(response.body)) {
             return last(response.body.type.split('/')) || 'default';
         }
 
         const error = last(response.body && response.body.errors);
 
-        return error && typeof error === 'object' && error.code ?
-            error.code :
-            'default';
+        return error ? error.code : 'default';
+    }
+
+    private _isStorefrontErrorResponseBody(
+        errorResponse: ErrorResponseBody
+    ): errorResponse is StorefrontErrorResponseBody {
+        const storefrontErrorResponse = errorResponse as StorefrontErrorResponseBody;
```

Comment:
I could be wrong, but I think you probably don't need to cast it as `StorefrontErrorResponseBody`.